python_modules/libraries/dagster-airbyte/dagster_airbyte_tests/experimental/test_resources.py

```diff
@@ -365,3 +371,80 @@ def test_airbyte_sync_and_poll_client_cancel_on_termination(
     assert_rest_api_call(
         call=base_api_mocks.calls[-1], endpoint=test_job_endpoint, method=last_call_method
     )
+
+
+def test_fivetran_airbyte_cloud_sync_and_poll_materialization_method(
+    fetch_workspace_data_api_mocks: responses.RequestsMock,
+    airbyte_cloud_sync_and_poll: MagicMock,
+    capsys: pytest.CaptureFixture,
+) -> None:
+    with environ(
+        {"AIRBYTE_CLIENT_ID": TEST_CLIENT_ID, "AIRBYTE_CLIENT_SECRET": TEST_CLIENT_SECRET}
+    ):
+        workspace = AirbyteCloudWorkspace(
+            workspace_id=TEST_WORKSPACE_ID,
+            client_id=EnvVar("AIRBYTE_CLIENT_ID"),
+            client_secret=EnvVar("AIRBYTE_CLIENT_SECRET"),
+        )
+
+        cleaned_connection_id = clean_name(TEST_CONNECTION_ID)
+
+        @airbyte_assets(
+            connection_id=TEST_CONNECTION_ID, workspace=workspace, name=cleaned_connection_id
+        )
+        def my_airbyte_assets(context: AssetExecutionContext, airbyte: AirbyteCloudWorkspace):
+            yield from airbyte.sync_and_poll(context=context)
+
+        # Mocked AirbyteCloudClient.sync_and_poll returns API response where all connection tables are expected
+        result = materialize(
+            [my_airbyte_assets],
+            resources={"airbyte": workspace},
+        )
+        assert result.success
+        asset_materializations = [
+            event
+            for event in result.events_for_node(cleaned_connection_id)
+            if event.event_type_value == "ASSET_MATERIALIZATION"
```

Comment:
nit: weird assertion. should assert against the actual dagster event type

Reply:
Comments addressed in [efd66b3](https://github.com/dagster-io/dagster/pull/26559/commits/efd66b3f2782485114d714c607442261c260e59a)